nerfstudio/fields/density_fields.py

```diff
@@ -66,6 +67,7 @@ def __init__(
         self.register_buffer("max_res", torch.tensor(max_res))
         self.register_buffer("num_levels", torch.tensor(num_levels))
         self.register_buffer("log2_hashmap_size", torch.tensor(log2_hashmap_size))
+        self.register_buffer("average_init_density", torch.tensor(average_init_density))
```

Comment:
Why is this a buffer here?

Reply:
This is my question as well, I saw other configurable parameters are set as buffer. I just tried to mimic the code style.

Anyone who can explain why `max_res`, `num_levels`, `log2_hashmap_size` are set as buffer?

Reply:
Seems a bit wasteful to me.

Reply:
Also it makes the new checkpoint not compatible with the previous one.

Reply:
It was originally changed here - https://github.com/nerfstudio-project/nerfstudio/pull/1486

Reply:
In the case of aabb, it was a parameter before. It makes sense as it is a tensor. For scalars I think it’s better to keep them as floats?

Reply:
Probably Volinga wanted it so they could read config values from the checkpoint's state_dict? But it also seems weird to me.